### PR TITLE
Promote Binary Serializer/Deserializer to IO

### PR DESCRIPTION
**Pull request by benharsh**
This PR promotes the BinarySerializer and BinaryDeserializer types to the IO module.

Testing:
- [x] full local

### CODE REVIEW

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -3043,6 +3043,389 @@ record DefaultDeserializer {
   }
 }
 
+/*
+  A basic binary Serializer to be used with ``fileWriter``.
```
Comment (jeremiah-corrado) [inline modules/standard/IO.chpl:3047]: I might remove the word "basic" here. It seems to have a slightly negative connotation.

Possibly replace with "direct"?

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -3043,6 +3043,389 @@ record DefaultDeserializer {
...
+    writer.write(val);
+  }
+
+  @chpldoc.nodoc
+  proc endMap(writer: _writeType) throws {
+  }
+}
+
+/*
+  A basic binary Deserializer to be used with ``fileReader``.
```
Comment (jeremiah-corrado) [inline modules/standard/IO.chpl:3214]: Ditto

**Review by jeremiah-corrado — APPROVED** [review]
Looks good to me! 

I just have one small wording suggestion:

File: modules/standard/IO.chpl (hunk 1 above)
Comment (benharsh) [reply modules/standard/IO.chpl:3047]: How about "unstructured"?

Comment (jeremiah-corrado) [reply modules/standard/IO.chpl:3047]: That sounds good 👍

**Pull request merged**